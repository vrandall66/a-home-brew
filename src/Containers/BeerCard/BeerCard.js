import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { bindActionCreators } from 'redux';
// import { FaBookmark } from 'react-icons/fa';
import { FaRegBookmark } from 'react-icons/fa';
import { IoIosBeer } from 'react-icons/io';
import {
  removeBookmark,
  setBookmark,
  setPreviousBrew,
  removePreviousBrew
} from '../../actions';
import './BeerCard.scss';

export class BeerCard extends Component {
  handleSaveClick = list => {
    const { id } = this.props.beer;
    switch (list) {
      case 'bookmark':
        this.toggleBookmark(id);
        break;
      case 'previous':
        this.togglePreviousBrew(id);
        break;
      // case 'current':
      //   toggleCurrentBrew(id);
      //   break;
      default:
        return null;
    }
  };

  toggleBookmark = id => {
    const { bookmarks, setBookmark, removeBookmark } = this.props;
    if (bookmarks.includes(id)) {
      removeBookmark(id);
    } else {
      setBookmark(id);
    }
  };

  togglePreviousBrew = id => {
    const { previousBrews, setPreviousBrew, removePreviousBrew } = this.props;
    if (previousBrews.includes(id)) {
      removePreviousBrew(id);
    } else {
      setPreviousBrew(id);
    }
  };

  render() {
    const { beer, type } = this.props;
    // const { beer, type, bookmarks, previousBrews } = this.props;
    // const isBookmarked = bookmarks.map(id => id === parseInt(beer.id));
    // const bookmarkImg = isBookmarked ? <FaBookmark /> : <FaRegBookmark />;
    // const bookmarkClass = isBookmarked ? 'bookmarked' : '';
    // console.log('anything?', isBookmarked);
    return (
      <div className='BeerCard' key={beer.id}>
        <header className='BeerCard__header'>
          <FaRegBookmark onClick={() => this.handleSaveClick('bookmark')} />
          <IoIosBeer onClick={() => this.handleSaveClick('previous')} />
          {/* <input
            type='image'
            src='images/Current.png'
            onClick={() => this.handleSaveClick('current')}
          /> */}
        </header>
        <h4 className='BeerCard__h4--name'>{beer.name}</h4>
        <p className='BeerCard__p--tagline'>{beer.tagline}</p>
        <Link to={`/${type}/${beer.id}`}>
          <button type='button' className='BeerCard__button--readMore'>
            Read More
          </button>
        </Link>
      </div>
    );
  }
}

export const mapStateToProps = ({ beers, bookmarks, previousBrews }) => ({
  beers,
  bookmarks,
  previousBrews
});

export const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      setBookmark,
      removeBookmark,
      setPreviousBrew,
      removePreviousBrew
    },
    dispatch
  );

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(BeerCard);
