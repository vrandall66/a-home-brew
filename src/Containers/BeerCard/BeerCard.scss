.BeerCard {
  justify-content: space-between;
  background-color: rgba(249, 225, 174, 0.8);
  display: flex;
  flex-direction: column;
  height: 150px;
  margin: 1.5%;
  max-width: 300px;
  padding: 1% 2%;
  width: 100%;
}

.BeerCard__header {
  display: flex;
  justify-content: space-between;
  padding: 2% 0%;
}

.BeerCard__h4--name {
  margin: 0;
}

.BeerCard__input--image {
  height: 16px;
  width: 16px;
}

.BeerCard__input--image-active {
  height: 16px;
  width: 16px;
}

.BeerCard__icon:hover {
  cursor: pointer;
}