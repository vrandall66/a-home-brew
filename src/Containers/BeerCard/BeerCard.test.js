import React from 'react';
import { shallow } from 'enzyme';
import {
  setBookmark,
  removeBookmark,
  setPreviousBrew,
  removePreviousBrew
} from '../../apiCalls/apiCalls';
import { Link } from 'react-router-dom';
import { FaBookmark, FaRegBookmark, FaPlus, FaCheck } from 'react-icons/fa';
import { BeerCard, mapStateToProps, mapDispatchToProps } from './BeerCard';
jest.mock('../../apiCalls/apiCalls');

describe('BeerCard', () => {
  let wrapper, wrapper2;

  beforeEach(() => {
    let beers = [{}, {}, {}];
    let bookmarks = [1, 2, 3];
    let previousBrews = [6, 7, 8];
    let beer = {
      id: 1,
      name: 'Buzz',
      tagline: 'A Real Bitter Experience.',
      first_brewed: '09/2007',
      description:
        'A light, crisp and bitter IPA brewed with English and American hops. A small batch brewed only once.',
      image_url: 'https://images.punkapi.com/v2/keg.png',
      abv: 4.5,
      ibu: 60,
      target_fg: 1010,
      target_og: 1044,
      ebc: 20,
      srm: 10,
      ph: 4.4,
      attenuation_level: 75,
      brewers_tips:
        'The earthy and floral aromas from the hops can be overpowering. Drop a little Cascade in at the end of the boil to lift the profile with a bit of citrus.',
      contributed_by: 'Sam Mason <samjbmason>'
    };
    let id = 1;
    let type = 'bookmark';
    let bookmark = true;
    let previous = false;

    wrapper = shallow(
      <BeerCard
        key={beer.id}
        beer={beer}
        type={type}
        bookmark={bookmark}
        previous={previous}
      />
    );

    wrapper2 = shallow(
      <div className='BeerCard' key={beer.id}>
        <header className='BeerCard__header'>
          {bookmark ? (
            <FaBookmark onClick={() => this.handleSaveClick('bookmark')} />
          ) : (
            <FaRegBookmark onClick={() => this.handleSaveClick('bookmark')} />
          )}
          {previous ? (
            <FaCheck onClick={() => this.handleSaveClick('previous')} />
          ) : (
            <FaPlus onClick={() => this.handleSaveClick('previous')} />
          )}
        </header>
        <h4 className='BeerCard__h4--name'>{beer.name}</h4>
        <p className='BeerCard__p--tagline'>{beer.tagline}</p>
        <Link to={`/${type}/${beer.id}`}>
          <button type='button' className='BeerCard__button--readMore'>
            Read More
          </button>
        </Link>
      </div>
    );
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call handleSaveClick onClick of bookmark icon', () => {
    wrapper.instance().handleSaveClick = jest.fn();
    wrapper.instance().forceUpdate();
    wrapper.find('FaBookmark').simulate('click');

    expect(wrapper.instance().handleSaveClick).toHaveBeenCalled();
  });

  it('should call handleSaveClick onClick of previous icon', () => {
    wrapper.instance().handleSaveClick = jest.fn();
    wrapper.instance().forceUpdate();
    wrapper.find('FaPlus').simulate('click');

    expect(wrapper.instance().handleSaveClick).toHaveBeenCalled();
  });
});
