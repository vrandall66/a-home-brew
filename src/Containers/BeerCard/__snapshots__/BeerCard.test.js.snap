// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BeerCard should match the snapshot 1`] = `
<div
  className="BeerCard"
  key="1"
>
  <header
    className="BeerCard__header"
  >
    <FaBookmark
      onClick={[Function]}
    />
    <FaPlus
      onClick={[Function]}
    />
  </header>
  <h4
    className="BeerCard__h4--name"
  >
    Buzz
  </h4>
  <p
    className="BeerCard__p--tagline"
  >
    A Real Bitter Experience.
  </p>
  <Link
    to="/bookmark/1"
  >
    <button
      className="BeerCard__button--readMore"
      type="button"
    >
      Read More
    </button>
  </Link>
</div>
`;
